File: src/QxRibbonQuickAccessBar.cpp
/**
 * Copyleft (C) 2023 maminjie <canpool@163.com>
 * SPDX-License-Identifier: MIT
**/
#include "QxRibbonQuickAccessBar.h"
#include "QxRibbonButtonGroup.h"

class RibbonQuickAccessBarPrivate
{
public:
    RibbonButtonGroup *m_buttonGroup;
};

RibbonQuickAccessBar::RibbonQuickAccessBar(QWidget *parent)
    : RibbonCtrlContainer(parent)
    , d(new RibbonQuickAccessBarPrivate)
{
    d->m_buttonGroup = new RibbonButtonGroup(this);
    d->m_buttonGroup->setObjectName(QStringLiteral("qx_RibbonButtonGroup"));
    setWidget(d->m_buttonGroup);
    setTitleVisible(false);
}

RibbonQuickAccessBar::~RibbonQuickAccessBar()
{
    delete d;
}

void RibbonQuickAccessBar::addSeparator()
{
    d->m_buttonGroup->addSeparator();
}

void RibbonQuickAccessBar::addAction(QAction *act)
{
    d->m_buttonGroup->addAction(act);
}

void RibbonQuickAccessBar::insertAction(QAction *before, QAction *action)
{
    d->m_buttonGroup->insertAction(before, action);
}

void RibbonQuickAccessBar::insertActions(QAction *before, QList<QAction *> actions)
{
    d->m_buttonGroup->insertActions(before, actions);
}

void RibbonQuickAccessBar::removeAction(QAction *action)
{
    d->m_buttonGroup->removeAction(action);
}

void RibbonQuickAccessBar::addWidget(QWidget *w)
{
    d->m_buttonGroup->addWidget(w);
}

void RibbonQuickAccessBar::addMenu(QMenu *m, QToolButton::ToolButtonPopupMode popMode)
{
    d->m_buttonGroup->addMenu(m, popMode);
}

File: pyqt/PyRibbon/QxRibbonQuickAccessBar.py
#!/usr/bin/env python3
# Copyright (c) 2023 maminjie <canpool@163.com>
# SPDX-License-Identifier: MIT


from PyQt5.QtWidgets import QAction, QWidget, QMenu, QToolButton

from .QxRibbonContainers import RibbonCtrlContainer
from .QxRibbonButtonGroup import RibbonButtonGroup


class RibbonQuickAccessBar(RibbonCtrlContainer):

    def __init__(self, parent=None):
        super().__init__(parent)
        self.m_buttonGroup = RibbonButtonGroup(self)
        self.setWidget(self.m_buttonGroup)

    def addSeparator(self):
        self.m_buttonGroup.addSeparator()

    def addAction(self, action: QAction):
        self.m_buttonGroup.addAction(action)

    def addWidget(self, w: QWidget):
        self.m_buttonGroup.addWidget(w)

    def addMenu(self, m: QMenu, pop_mode=QToolButton.InstantPopup):
        self.m_buttonGroup.addMenu(m, pop_mode)
